### ISSUE TITLE
[HDSky] Error downloading. Remote server returned NotFound.

### ISSUE THREAD

**Issue by JasonL708** [Jackett/Jackett#14923]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Using HTTP Client: HttpWebClient2
Using FlareSolverr: http://flaresolverr:8191/
Using proxy: Disabled
App config/log directory: /config/Jackett
ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads
Running in Docker: Yes (image build: v0.21.1396-ls250)
File /etc/issue: Welcome to Alpine Linux 3.18
Jackett variant: CoreLinuxMuslAmdx64
OS version: Unix 4.4.302.0 (64bit OS) (64bit process)
Environment version: 6.0.25 (/app/Jackett/)
Starting Jackett v0.21.1424
```

### Description

In Sonarr (Same as Radarr) Interactive Search interface, I can see the list of torrents. However, when I click the download button, it says "Getting release from indexer fail".

From my limited understanding, seems like the download path generated by Jackett is different from what I got from the website. The path generated is 
 `https://hdsky.me/download.php?id=xx&t=xxx&sign=xxx`. The download link I get from the website directly looks like this: `https://hdsky.me/download.php?id=xxx&passkey=xxx&sign=xx`

### Logged Error Messages

```text
Error downloading. indexer: hdsky path: https://hdsky.me/download.php?id=(removed)&t=(Removed)&sign=(Removed)
System.Exception: Remote server returned NotFound
   at Jackett.Common.Indexers.BaseWebIndexer.Download(Uri link, RequestType method, String referer, Dictionary`2 headers) in ./Jackett.Common/Indexers/BaseIndexer.cs:line 521
   at Jackett.Common.Indexers.CardigannIndexer.Download(Uri link) in ./Jackett.Common/Indexers/CardigannIndexer.cs:line 2107
   at Jackett.Server.Controllers.DownloadController.DownloadAsync(String indexerId, String path, String jackett_apikey, String file) in ./Jackett.Server/Controllers/DownloadController.cs:line 56

Failed download response:
<h1>Not Found</h1>
<script>(function(){var js = "window['__CF$cv$params']={r:'83788dd37fe80465',t:'MTcwMjkxNDU0Ny45NzcwMDA='};_cpo=document.createElement('script');_cpo.nonce='',_cpo.src='/cdn-cgi/challenge-platform/scripts/jsd/main.js',document.getElementsByTagName('head')[0].appendChild(_cpo);";var _0xh = document.createElement('iframe');_0xh.height = 1;_0xh.width = 1;_0xh.style.position = 'absolute';_0xh.style.top = 0;_0xh.style.left = 0;_0xh.style.border = 'none';_0xh.style.visibility = 'hidden';document.body.appendChild(_0xh);function handler() {var _0xi = _0xh.contentDocument || _0xh.contentWindow.document;if (_0xi) {var _0xj = _0xi.createElement('script');_0xj.innerHTML = js;_0xi.getElementsByTagName('head')[0].appendChild(_0xj);}}if (document.readyState !== 'loading') {handler();} else if (window.addEventListener) {document.addEventListener('DOMContentLoaded', handler);} else {var prev = document.onreadystatechange || function () {};document.onreadystatechange = function (e) {prev(e);if (document.readyState !== 'loading') {document.onreadystatechange = prev;handler();}};}})();</script>
```

**Comment by garfield69** [comment]
The indexer currently uses the download link from the torrents.php page. Its possible the site has updated their web site.
However, since none of the current team have an account on hdsky we cannot independently research.
Please use your browser to access the hdsky torrents.php and provide the HTML, see https://github.com/Jackett/Jackett/wiki/Troubleshooting/#how-to-provide-html-source
Its possible that the web site has had an update and we need to make changes to the Jackett indexer.
Thanks.

**Comment by JasonL708** [comment]
[HDSky __ Search results for Game of Thrones 高清视界_高清天空_高清发烧友的后花园 - Powered by NexusPHP.txt](https://github.com/Jackett/Jackett/files/13715727/HDSky.__.Search.results.for.Game.of.Thrones._._.-.Powered.by.NexusPHP.txt)

Please see the file above. Hope that this helps and feel free to let me know if you need anything further. 

Thanks.

**Comment by garfield69** [comment]
mynameisbogdan may I ask why you recommend changing the passkey, is it because you think its been compromised someplace? I don't see anything in the logs or html dump.
Right now hdsky is in a login lockdown, so anything that might end up requiring a login is going to cause an issue. I do not know if changing passkey will fall into this category, just mentioning it as a possible complication.

**Comment by garfield69** [comment]
@JasonL708  could you provide an HTML dump of a torrent details page please, so we can check if the download link on that page is different from the one on the torrents search page.
Thanks.

**Comment by JasonL708** [comment]
[HDSky __ Details for torrent _Silent Night 2023 2160p WEB-DL DDP5.1 Atmos DV HDR H.265-Fury@HDSky_ 高清视界_高清天空_高清发烧友的后花园 - Powered by NexusPHP.txt](https://github.com/Jackett/Jackett/files/13740746/HDSky.__.Details.for.torrent._Silent.Night.2023.2160p.WEB-DL.DDP5.1.Atmos.DV.HDR.H.265-Fury%40HDSky_._._.-.Powered.by.NexusPHP.txt)

@garfield69 thank you for the passkey rest reminder. I will contact the forum operator to reset my passkey.

Please see above for the HTML dump of torrent detail page. So in this HTML you can find the `https://hdsky.me/download.php?id=xxx&passkey=xxx&sign=xx` link that I mentioned earlier. Via that link, I can directly download the torrent document.

In addition, there are two other links. One looks identical to what Jackett would generate as `https://hdsky.me/download.php?id=xxx&t=xxx&sign=xxx`, and another one with an addition parament of `type=zip`. Both links work, except that the second one obviously will download a Zip file. 

Thanks.

**Comment by JasonL708** [comment]
Hi guys, I realized that I was being stupid. The reason why it didn't work is that I have stayed at the interactive search result page for too long before I hit the download button. In that case, the link generated by the download button has already expired and that is the reason why the download could work. I have re-tried it today after refreshing the page and everything works normally. Sorry to bother you guys and thanks a lot for all the help!
